### ISSUE TITLE
pass method return value as positional parameters to Response

### ISSUE THREAD

**Issue by dmig-alarstudios** [tiangolo/fastapi#1628]
Recently I've noticed that `status_code` parameter of the next statement is useless:
```python3
@router.get('/callback/{provider}', status_code=303,
            response_description='Redirect to the application login',
            response_class=RedirectResponse)
```
Because I always have to create `RedirectResponse` objects manually:
```python3
        return RedirectResponse(target.include_query_params(error=f'{provider}_{error_reason}'),
                                status_code=303)
```

I've tried to play with return values and noticed that return value is always passed to `response_class` instance as a `content` parameter:
```python3
@app.get('/test', status_code=303, response_class=RedirectResponse)
async def test():
    return '/'
```
this snippet produces the next exception:
```
INFO:     127.0.0.1:43878 - "GET /test HTTP/1.1" 500 Internal Server Error
ERROR:    Exception in ASGI application
Traceback (most recent call last):
  File "/home/dmig/.pyenv/versions/fastapi-sandbox/lib/python3.8/site-packages/uvicorn/protocols/http/httptools_impl.py", line 385, in run_asgi
    result = await app(self.scope, self.receive, self.send)
  File "/home/dmig/.pyenv/versions/fastapi-sandbox/lib/python3.8/site-packages/uvicorn/middleware/proxy_headers.py", line 45, in __call__
    return await self.app(scope, receive, send)
  File "/home/dmig/.pyenv/versions/fastapi-sandbox/lib/python3.8/site-packages/fastapi/applications.py", line 171, in __call__
    await super().__call__(scope, receive, send)
  File "/home/dmig/.pyenv/versions/fastapi-sandbox/lib/python3.8/site-packages/starlette/applications.py", line 102, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/home/dmig/.pyenv/versions/fastapi-sandbox/lib/python3.8/site-packages/starlette/middleware/errors.py", line 181, in __call__
    raise exc from None
  File "/home/dmig/.pyenv/versions/fastapi-sandbox/lib/python3.8/site-packages/starlette/middleware/errors.py", line 159, in __call__
    await self.app(scope, receive, _send)
  File "/home/dmig/.pyenv/versions/fastapi-sandbox/lib/python3.8/site-packages/starlette/exceptions.py", line 82, in __call__
    raise exc from None
  File "/home/dmig/.pyenv/versions/fastapi-sandbox/lib/python3.8/site-packages/starlette/exceptions.py", line 71, in __call__
    await self.app(scope, receive, sender)
  File "/home/dmig/.pyenv/versions/fastapi-sandbox/lib/python3.8/site-packages/starlette/routing.py", line 550, in __call__
    await route.handle(scope, receive, send)
  File "/home/dmig/.pyenv/versions/fastapi-sandbox/lib/python3.8/site-packages/starlette/routing.py", line 227, in handle
    await self.app(scope, receive, send)
  File "/home/dmig/.pyenv/versions/fastapi-sandbox/lib/python3.8/site-packages/starlette/routing.py", line 41, in app
    response = await func(request)
  File "/home/dmig/.pyenv/versions/fastapi-sandbox/lib/python3.8/site-packages/fastapi/routing.py", line 217, in app
    response = response_class(
TypeError: __init__() got an unexpected keyword argument 'content'
```

I didn't dive into the code yet, so this is not a PR but a request. But the idea is simple:
```python3
response  = response_class(*(endpoint_result if isinstance(endpoint_result, (Tuple, List)) else tuple(endpoint_result)))
```
Or maybe even more complex logic: pass `**endpoint_result` if `endpoint_result` is a `Mapping`, pass `*endpoint_result` if it is an `Iterable` or else pass it as a `*tuple(endpoint_result)`

**Comment by dmig-alarstudios** [comment]
as a simpler alternative -- just add `content` alias to `url` for `RedirectResponse`

**Comment by dmig** [comment]
To me, this seems to be a design flaw in Starlette...

**Comment by tiangolo** [comment]
Thanks! That makes sense. I updated the code internally to allow returning the redirect URL directly, also paths for `FileResponse`, etc. It was implemented here: https://github.com/tiangolo/fastapi/pull/3457

It will be available tomorrow on FastAPI version `0.56.0` :tada: :rocket:
